artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQMessage.java

```diff
@@ -853,10 +853,14 @@ public boolean waitCompletionOnStream(final long timeWait) throws JMSException {
    @Override
    public String toString() {
       StringBuffer sb = new StringBuffer("ActiveMQMessage[");
-      sb.append(getJMSMessageID());
-      sb.append("]:");
-      sb.append(message.isDurable() ? "PERSISTENT" : "NON-PERSISTENT");
-      sb.append("/" + message.toString());
+      if (message != null) {
```

Comment:
I agree it is reasonably but I see an empty constructor.

Reply:
Thats a fair point, but my question then becomes what is using it and can it go.

Reply:
The constructor only looks to be directly called by some 'compatible' subclasses empty constructors (which then arent called directly), and some tests that could easily use one of the several other constructors. Everything operates on the delegate ClientMessage instance passed or created in all the other constructors, thats actually where I'd expect decoding happens rather than here.

I agree there is no real harm to 'fixing' it this way, I just think its generally changing the wrong thing and leaving an issue that neednt exist. Everything else here clearly expects its non-null, enforcing that seems the appropriate way to go.